File: modules/ballerina-core/src/main/java/org/wso2/ballerina/core/model/expressions/KeyValueExpression.java
/*
 * Copyright (c) 2016, WSO2 Inc. (http://wso2.com) All Rights Reserved.
 * <p>
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.wso2.ballerina.core.model.expressions;

import org.wso2.ballerina.core.model.NodeVisitor;

/**
 * Class to hold map initialization data
 */
public class KeyValueExpression extends AbstractExpression {
    String key;
    Expression valueExpression;

    public KeyValueExpression(String key, Expression valueExpression) {
        this.key = key;
        this.valueExpression = valueExpression;
    }
    @Override
    public void accept(NodeVisitor visitor) {
        visitor.visit(this);
    }
    public String getKey() {
        return key;
    }
    public Expression getValueExpression() {
        return valueExpression;
    }
}

File: modules/web/js/ballerina/ast/key-value-expression.js
/**
 * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
define(['lodash', './expression'], function (_, Expression) {

    /**
     * Constructor for KeyValueExpression
     * @param {Object} args - Arguments to create the KeyValueExpression
     * @constructor
     */
    var KeyValueExpression = function (args) {
        Expression.call(this, 'KeyValueExpression');
        this._key = _.get(args, 'key', '');
        this._expression = _.get(args, 'valueExpression', '');
    };

    KeyValueExpression.prototype = Object.create(Expression.prototype);
    KeyValueExpression.prototype.constructor = KeyValueExpression;

    /**
     * setting parameters from json
     * @param jsonNode
     */
    KeyValueExpression.prototype.initFromJson = function (jsonNode) {
        this.setExpression(this.generateKeyValueExpression(jsonNode), {doSilently: true});
    };

    /**
     * function for generate string for key-value-expression
     * @param sonNode
     */
    KeyValueExpression.prototype.generateKeyValueExpression = function (jsonNode) {
        var expString = "";
        var self = this;
        var keyExpressionNode = self.getFactory().createFromJson(jsonNode.children[1][0]);
        var valueExpressionNode = self.getFactory().createFromJson(jsonNode.children[0]);
        valueExpressionNode.initFromJson(jsonNode.children[0]);
        keyExpressionNode.initFromJson(jsonNode.children[1][0]);
        expString += keyExpressionNode.getExpression() + ":" + valueExpressionNode.getExpression();
        return expString;
    };
    return KeyValueExpression;
});
